lib/rollup/query/agent_ranks.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Castle.Rollup.Query.AgentRanks do
+  import Ecto.Query
+  alias Castle.Rollup.Query.AgentTotals, as: AgentTotals
+
+  def podcast(id, %{from: from, to: to, bucket: bucket}, %{name: grouping_name, limit: num}) do
+    podcast(id, from, to, bucket.rollup, grouping_name, num)
+  end
+  def podcast(id, from, to, trunc, grouping_name, grouping_limit) do
+    top_n = AgentTotals.podcast(id, from, to, grouping_name, grouping_limit) |> Enum.map(&(&1.group))
```

Comment:
This was driving me nuts.  Couldn't find ~a good~ _any_ way to make Ecto do this subquery.  So I just ask the Totals query for the "top N groups for this podcast, over this period of time".